user_manager/signals.py

```diff
@@ -0,0 +1,34 @@
+"""
+Signals for User Manager app.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from .models import UserManagerRole
+
+
+def _upgrade_manager_role_entry(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Upgrade an email link to a proper FK link when a manager registers.
+    """
+    created = kwargs.get('created')
+    user = kwargs.get('instance')
+
+    if created and user:
+        UserManagerRole.objects.filter(
+            unregistered_manager_email=user.email
+        ).update(
+            unregistered_manager_email=None,
+            manager_user=user,
+        )
+
+
+@receiver(post_save, sender=User)
+def upgrade_manager_role_entry(sender, **kwargs):
+    """
+    Call function to upgrade user-manager link to proper FK.
+    """
+    _upgrade_manager_role_entry(sender, **kwargs)
```

Comment:
Nit: You don't need to pass `sender`, and this function might be nicer and more useful if it extracts the `created` and `user` fields from `kwargs` and just passes those two values to `_upgrade_manager_role_entry` ? Then you can also remove the `# pylint: disable=unused-argument` above.

Reply:
Yup, of course! Silly me, I just extracted this into a separate function to make it testable, but for some reason didn't think of this.